releng_sop/pulp_clone_repos.py

```diff
@@ -0,0 +1,365 @@
+        :type force:     Boolean
+
+        :param commit:   Flag to indicate if password is required for a commit action.
+                         If not set, password will not be asked for.
+        :type commit:    Boolean
+        """
+        if not commit:
+            return
+
+        result = self.pulp_password
+
+        if force:
+            result = ""
+
+        while not result:
+            msg = "Enter Pulp password for %s@%s: "
+            prompt = msg % (self.pulp_config["client"]["user"], self.pulp_config.name)
+            result = getpass.getpass(prompt=prompt)
+
+        self.pulp_password = result
+
+    def query_repo(self):
+        """Get name of pdc repo_from and pdc repo_to."""
+        if self.query_data_from['release_id'] == self.query_data_to['release_id']:
+            raise UsageError('Release id is same')
+        client = PDCClient(self.env["pdc_server"], develop=True)
+
+        result_from = client['content-delivery-repos']._(page_size=0, **self.query_data_from)
+        result_to = client['content-delivery-repos']._(page_size=0, **self.query_data_to)
+
+        if (len(result_from) != len(result_to)) and (not self.skip_repo_check):
+            raise UsageError('Error')
+
+        rep_from = self.rearange(result_from)
+        rep_to = self.rearange(result_to)
+
+        self.cloned = []
+        self.sameName = []
+        self.missDest = []
+        self.missSource = []
+
+        while len(rep_from):
+            map_key, name = rep_from.popitem()
+            if map_key in rep_to:
+                if name == rep_to[map_key]:
+                    self.sameName.append({'from': name, 'to': rep_to[map_key]})
+                else:
+                    self.cloned.append({'from': name, 'to': rep_to[map_key]})
+                del rep_to[map_key]
+            else:
+                self.missDest.append({'from': name})
+        while len(rep_to):
+            map_key, name = rep_to.popitem()
+            self.missSource.append({'to': name})
+        self.cloned = sorted(self.cloned, key=lambda x: sorted(x.values()))
+
+    def details(self, commit=False):
+        """
+        Print details of command execution.
+
+        :param commit: Flag to indicate if the command will be actually executed.
+                       Line indicating "test mode" is printed, if this is False.
+        :type  commit: boolean=False
+        """
+        self.query_repo()
+
+        details = "Pulp clone repos\n"
+        details += " * env name:                %s\n" % self.env.name
+        details += " * env config:              %s\n" % self.env.config_path
+        details += " * release source           %s\n" % self.release_from.config_path
+        details += " * PDC server:              %s\n" % self.env["pdc_server"]
+        details += " * release_id from:         %s\n" % self.release_id_from
+        details += " * release_id to:           %s\n" % self.release_id_to
+        if self.content_category:
```

Comment:
`content_category` should be as `arches` and `variants`, a similar mechanism could be used to print the selected *content categories*, if there are any.